From 8410512d92437d60a15760cde8508f5c26fdb99d Mon Sep 17 00:00:00 2001
From: Michael Krummsdorf <michael.krummsdorf@tq-group.com>
Date: Wed, 24 Jan 2018 10:27:51 +0100
Subject: [PATCH 047/188] ehci-fsl: add arm equivalent for clrsetbits and
 clrbits

These are powerpc definitions which fail on arm compilations.
For use with LS1021A define the ARM specific pendant.

Signed-off-by: Michael Krummsdorf <michael.krummsdorf@tq-group.com>
---
 arch/arm/include/asm/io.h   | 4 ++++
 drivers/usb/host/ehci-fsl.c | 1 +
 2 files changed, 5 insertions(+)

diff --git a/arch/arm/include/asm/io.h b/arch/arm/include/asm/io.h
index 2cfbc531f63b..6ad0784b9768 100644
--- a/arch/arm/include/asm/io.h
+++ b/arch/arm/include/asm/io.h
@@ -431,6 +431,10 @@ void *arch_memremap_wb(phys_addr_t phys_addr, size_t size);
 
 #define iowrite16be(v,p)	({ __iowmb(); __raw_writew((__force __u16)cpu_to_be16(v), p); })
 #define iowrite32be(v,p)	({ __iowmb(); __raw_writel((__force __u32)cpu_to_be32(v), p); })
+#define clrsetbits_be32(addr, clear, set) iowrite32be((ioread32be((addr)) & ~(clear)) | (set), (addr))
+#define clrbits32(_addr, _v)	iowrite32be(ioread32be(_addr) & ~(_v), (_addr))
+#define mfspr(x)		0
+#define SPRN_SVR		0
 
 #ifndef ioport_map
 #define ioport_map ioport_map
diff --git a/drivers/usb/host/ehci-fsl.c b/drivers/usb/host/ehci-fsl.c
index d025cc06dda7..2ef839ab486c 100644
--- a/drivers/usb/host/ehci-fsl.c
+++ b/drivers/usb/host/ehci-fsl.c
@@ -25,6 +25,7 @@
 
 #include <linux/kernel.h>
 #include <linux/module.h>
+#include <linux/io.h>
 #include <linux/types.h>
 #include <linux/delay.h>
 #include <linux/pm.h>
